### DISCUSSION TITLE
How to set user in room?

### DISCUSSION THREAD

**Discussion by Zhukek** [strapi-community/plugin-io#47] [Q&A]
I have some categories of users, and i want to set them in specific rooms. Can't understand, how can i do this. I've tried something like in original socket.io, but failed:

`events:[
        {
          name: "connection",
          handler: ({ strapi }, socket) => {
            console.log('--connect--')
          },
        },
        {
          name: "setRoom",
          handler: ({ strapi }, socket) => {
            socket.join("Test room")
          },
        },
      ]`

**Comment by ComfortablyCoding** [comment]
Any custom events (e.g. setRoom) need to be emitted by you to be triggered. 

In terms of adding users to a room as an example you can do so on the connection event with the following code
```js
{
// ..
  events: [
    {
      name: "connection",
      handler: ({ strapi }, socket) => {
        console.log("--connect--");

        const room = "some-room";
        socket.join(room);
        console.log(`${socket.id} has joined room ${room}`);
      },
    },
  ];
// ..
}

```
